code/api/decorators.py

```diff
@@ -23,3 +25,21 @@ def wrapper_func(request, author_id, *args, **kwargs):
         return view_func(request, author_id, *args, **kwargs)
 
     return wrapper_func
+
+def validate_node(view_func):
+    """
+        Restrict to allowed nodes
+    """
+    def wrapper_func(request, *args, **kwargs):
+
+        auth_header = request.META.get('HTTP_AUTHORIZATION', '')
+        token_type, _, receivedCredentials = auth_header.partition(' ')
+
+        expectedCredentials = base64.b64encode(b'remotegroup:topsecret!').decode()
+
+        if not node_manager.get_host_credentials(request.META['HTTP_HOST']) or token_type != 'Basic' or receivedCredentials != expectedCredentials:
```

Comment:
`request.META['HTTP_HOST']` returns the host of the current server, not the host of the remote server who just made the request. If I only add the remote server (and not the local one) to Node in the admin panel, I get a 401.

Reply:
very good catch,
Not sure why it returns the incorrect host. trying to find a fix
Could it be a development server-only issue?

Reply:
I think we would have to require whoever made the request to add a referer header to their request.